core/domain/subtopic_validators_test.py

```diff
@@ -0,0 +1,858 @@
+        self.run_job_and_check_output(
+            expected_output, sort=True, literal_eval=False)
+
+
+class SubtopicPageSnapshotMetadataModelValidatorTests(
+        test_utils.AuditJobsTestBase):
+
+    def setUp(self):
+        super(SubtopicPageSnapshotMetadataModelValidatorTests, self).setUp()
+        self.signup(self.OWNER_EMAIL, self.OWNER_USERNAME)
+        self.owner_id = self.get_user_id_from_email(self.OWNER_EMAIL)
+
+        self.signup(USER_EMAIL, USER_NAME)
+        self.user_id = self.get_user_id_from_email(USER_EMAIL)
+
+        self.signup(self.ADMIN_EMAIL, self.ADMIN_USERNAME)
+        self.admin_id = self.get_user_id_from_email(self.ADMIN_EMAIL)
+        self.set_admins([self.ADMIN_USERNAME])
+
+        topics = [topic_domain.Topic.create_default_topic(
+            '%s' % i,
+            'topic%s' % i,
+            'abbrev-%s' % chr(120 + i),
+            'description%s' % i) for i in python_utils.RANGE(3)]
+        language_codes = ['ar', 'en', 'en']
+
+        for index, topic in enumerate(topics):
+            topic.language_code = language_codes[index]
+            topic.add_additional_story('%s' % (index * 2))
+            topic.add_canonical_story('%s' % (index * 2 + 1))
+            topic.add_uncategorized_skill_id('%s' % (index * 3))
+            topic.add_uncategorized_skill_id('%s' % (index * 3 + 1))
+            topic.add_uncategorized_skill_id('%s' % (index * 3 + 2))
+            topic_services.save_new_topic(self.owner_id, topic)
+            if index == 0:
+                committer_id = self.user_id
+            else:
+                committer_id = self.owner_id
+            topic_services.update_topic_and_subtopic_pages(
+                committer_id, '%s' % index, [topic_domain.TopicChange({
+                    'cmd': 'add_subtopic',
+                    'title': 'subtopic1',
+                    'subtopic_id': 1
+                }), topic_domain.TopicChange({
+                    'cmd': 'move_skill_id_to_subtopic',
+                    'old_subtopic_id': None,
+                    'new_subtopic_id': 1,
+                    'skill_id': '%s' % (index * 3)
+                }), topic_domain.TopicChange({
+                    'cmd': 'move_skill_id_to_subtopic',
+                    'old_subtopic_id': None,
+                    'new_subtopic_id': 1,
+                    'skill_id': '%s' % (index * 3 + 1)
+                })], 'Changes.')
+
+        self.model_instance_0 = (
+            subtopic_models.SubtopicPageSnapshotMetadataModel.get_by_id(
+                '0-1-1'))
+        self.model_instance_1 = (
+            subtopic_models.SubtopicPageSnapshotMetadataModel.get_by_id(
+                '1-1-1'))
+        self.model_instance_2 = (
+            subtopic_models.SubtopicPageSnapshotMetadataModel.get_by_id(
+                '2-1-1'))
+
+        self.job_class = (
+            prod_validation_jobs_one_off
+            .SubtopicPageModelAuditOneOffJob)
```

Comment:
Here you're working with other class

```suggestion
        self.job_class = (
            prod_validation_jobs_one_off
            .SubtopicPageSnapshotMetadataModelAuditOneOffJob)
```